withoutUndo with opt out

scytacki opened the pull request:
This fixes warnings that happened when the resizing of a row was being triggered by parent action.

In these cases it seems to make sense to just go ahead and record the resizing. The reason we didn't record it before because it could happen on document load and would result in a weird undo history.

The change really should just remove a warning that is printed in some cases. The adding of undo history events should behave exactly the same as when the warning was there.

scytacki commented:
I ran the failing cypress tests locally and they passed.

bgoldowsky reviewed (approved):
Code changes look straightforward and good.

This might be a good time to add JSDoc comments to these methods.

Also on the naming question, I noticed that there is an `onAnyAction` utility which takes an option called `onlyOuter` with, I think, the same semantics.  `unlessChildAction` seems a bit more clear, though.

scytacki commented:
Good point @bgoldowsky I added some JSDoc.

scytacki commented:
Something weird happened with the regression tests here. The ran on the first commit when I added the `run regression` label. Then when I added a second commit and quickly force pushed a revision to that second commit, the regression tests did not run. Because my only change was to add a comment it doesn't seem worth getting the regression tests to run again.

@tejal-shah I don't think this needs QA review. It is just removing a warning message printed in the console.
